internal/pkg/manifest/env.go

```diff
@@ -0,0 +1,100 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package manifest provides functionality to create Manifest files.
+package manifest
+
+import (
+	"sort"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/copilot-cli/internal/pkg/template"
+)
+
+// Environment is the manifest configuration for an environment.
+type Environment struct {
+	Workload          `yaml:",inline"`
+	environmentConfig `yaml:",inline"`
+
+	parser template.Parser
+}
+
+type environmentConfig struct {
+	Network environmentNetworkConfig `yaml:"network"`
+}
+
+type environmentNetworkConfig struct {
+	VPC environmentVPCConfig `yaml:"vpc"`
+}
+
+type environmentVPCConfig struct {
+	ID      *string              `yaml:"id"`
+	CIDR    *IPNet               `yaml:"cidr"`
+	Subnets subnetsConfiguration `yaml:"subnets"`
+}
+
+func (v environmentVPCConfig) imported() bool {
+	return aws.StringValue(v.ID) != ""
+}
+
+func (v environmentVPCConfig) managedVPCCustomized() bool {
+	return aws.StringValue((*string)(v.CIDR)) != ""
+}
+
+func (v environmentVPCConfig) ImportedVPC() *template.ImportVPC {
+	if !v.imported() {
+		return nil
+	}
+	var publicSubnetIDs, privateSubnetIDs []string
+	for _, subnet := range v.Subnets.Public {
+		publicSubnetIDs = append(publicSubnetIDs, aws.StringValue(subnet.SubnetID))
+	}
```

Comment:
```suggestion
	publicSubnetIDs := aws.StringValueSlice(v.Subnets.Public)
```

Reply:
same below

Reply:
not super applicable because we are extracting the `subnet.SubnetID` field. `v.Subnets.Public` itself is a list of struct :(

Reply:
ahhh you're right😊